src/AWS.Deploy.Common/Recommendation.cs

```diff
@@ -200,10 +200,35 @@ public bool IsOptionSettingDisplayable(OptionSettingItem optionSetting)
                 var dependsOnOptionSettingValue = GetOptionSettingValue(dependsOnOptionSetting);
                 if (
                     dependsOnOptionSetting != null &&
-                    dependsOnOptionSettingValue != null &&
-                    !dependsOnOptionSettingValue.Equals(dependency.Value))
+                    dependsOnOptionSettingValue != null)
                 {
-                    return false;
+                    if (dependency.Operation == null ||
+                           dependency.Operation == PropertyDependencyOperationType.Equals)
+                    {
+                        if (!dependsOnOptionSettingValue.Equals(dependency.Value))
+                            return false;
+                    }
+                    else if (dependency.Operation == PropertyDependencyOperationType.NotEmpty)
+                    {
+                        if (dependsOnOptionSetting.Type == OptionSettingValueType.List &&
+                            dependsOnOptionSettingValue != null &&

```

Comment:
If the list has never been set then `dependsOnOptionSettingValue` will be null. That should count as being empty and we should return `false`. But the way I read this logic the `dependsOnOptionSettingValue != null` condition will cause it to skip the return `false`.

Reply:
The condition on line 203 https://github.com/aws/aws-dotnet-deploy/blob/842ae771dc31933231b30fe19abcadc269ca3534/src/AWS.Deploy.Common/Recommendation.cs#L203 makes sure that `dependsOnOptionSettingValue` is not null. In any case, `dependsOnOptionSettingValue` is never null because of https://github.com/aws/aws-dotnet-deploy/blob/842ae771dc31933231b30fe19abcadc269ca3534/src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs#L53 
`dependsOnOptionSettingValue` will be an empty string which will be handled in the if condition.

Reply:
But if `dependsOnOptionSettingValue` is null then the condition on line 203 will cause it to skip processing either the `Equals` or `NotEmpty` operations and then return true. Sounds like we are getting lucky now because we always return back at least an empty string but that is frail. Especially since we even do a null check in this method on a property that we say should never be null.

Reply:
fixed